### ISSUE TITLE
Uncaught NPE when parsing a header value

### ISSUE THREAD

**Issue by nigredo-tori** [apache/pekko-http#574]
I'm using Akka 10.2.9, but the issue (as in, broken error handling) doesn't seem to have been fixed in Pekko as well. Sorry I don't have a reproduction example at this moment, I'm creating this issue to have something to reference when rolling out a patch that will hopefully give me a clue. :smile: 

I have a piece of code like this:
```scala
`Content-Disposition`.parseFromValueString(headerValue)
```

I expect it to return an `Either`. Instead, for some input values it seems to throw the following exception:
```
java.lang.NullPointerException: Cannot invoke "String.split(String, int)" because "message" is null
	at akka.http.scaladsl.model.ErrorInfo$.fromCompoundString(ErrorInfo.scala:65)
	at akka.http.impl.model.parser.HeaderParser.failure(HeaderParser.scala:77)
	at akka.http.scaladsl.model.HttpHeader$.parse(HttpHeader.scala:92)
	at akka.http.scaladsl.model.headers.ModeledCompanion.parseFromValueString(headers.scala:40)
        <redacted>
```
It seems that there is an exception raised somewhere inside the parser that doesn't have its message set. [Here](https://github.com/akka/akka-http/blob/v10.2.9/akka-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala#L77) we do `.getMessage`, passing `null` into `ErrorInfo.fromCompoundString`. It [tries to split it](https://github.com/akka/akka-http/blob/v10.2.9/akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala#L65), and we get an NPE.

**Comment by jrudolph** [comment]
Thanks for the report. `HeaderParser.failure` tries exactly to convert exceptions into an `Either`-like error but misses the fact that exception messages can be null. We could either return an empty ErrorInfo or try to be a bit more informative and infer some info from the exception class name.

**Comment by laglangyue** [comment]
<img width="909" alt="image" src="https://github.com/apache/pekko-http/assets/35491928/c9a59c73-55fd-4d17-a4d9-3c1b9ca9a290">

we should add case

**Comment by pjfanning** [comment]
> <img alt="image" width="909" src="https://private-user-images.githubusercontent.com/35491928/346989795-c9a59c73-55fd-4d17-a4d9-3c1b9ca9a290.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1MzQ3NzUsIm5iZiI6MTcyMDUzNDQ3NSwicGF0aCI6Ii8zNTQ5MTkyOC8zNDY5ODk3OTUtYzlhNTljNzMtNTVmZC00ZDE3LWE0ZDktM2MxYjljYTlhMjkwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA5VDE0MTQzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxYTE3NTFlN2UwYjNhYWMyZjQ4NzEyNTMxODJiNmQxMTc3NmU2M2RkNzlkYTIwYTQ5MjhkNjQ1NDBjOTM2OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.l2TCIxKiDO01IUjVaA_uA1d8o6Oda9iMzmvtJGor8Gc">
> we should add case

@laglangyue would you have time to do a PR?

**Comment by laglangyue** [comment]
@pjfanning yes,  I am designing some tests to reproduce errors and make modifications

**Comment by laglangyue** [comment]
Although I added a check that can fix it, the message of NPE is not null on Java17.

I dig deep again,
<img width="1328" alt="image" src="https://github.com/apache/pekko-http/assets/35491928/9a4c7686-260b-4eff-8020-45216eec9b01">
<img width="931" alt="image" src="https://github.com/apache/pekko-http/assets/35491928/c1f231fc-96f5-4da0-8c01-84506a1efc84">

I hope to identify NPE in advance, so I still need some time to read the code in the parse section

**Comment by pjfanning** [comment]
Use the getMessage if is not null even if NullPointerException has changed in Java 17. We support Java 8+.
Text cases may not to be handle supporting different behaviour depending on the Java Runtime Version being used.
With StringBasedParserInput, you can change the code that creates instances of StringBasedParserInput to pass in the exception toString instead of getMessage when the getMessage returns null or empty string. Use getMessage when there is a non-empty string (to retain compatibility with existing behaviour).

**Comment by pjfanning** [comment]
I'd like to see the fix backported to the 1.0.x branch.

**Comment by pjfanning** [comment]
PRs merged to main and 1.0.x branches
